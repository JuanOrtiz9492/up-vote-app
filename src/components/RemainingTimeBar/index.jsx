import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { cssConstants } from '../../config/cssConstants'

const { pearlRiver, white, coal } = cssConstants.colors;
const BarContainer = styled.div`
    display: flex;
    width: 100%;
    heigt: 100%;
    background rgba(${pearlRiver.r}, ${pearlRiver.g}, ${pearlRiver.b}, 0.6);
`
const ProgresBar = styled.div`
    display: flex;
    width: ${props => props.value}%;
    height: 100%;
    background: rgba(0,0,0, 0.2);
    position: relative;
    justify-content: flex-end;
    z-index:1;
    &::before{
        position: absolute;
        top: 35%;
        right:-0.8em;
        content: '';
        width: 0;
        height: 0;
        border-top: 0.5em solid transparent;
        border-left: 0.8em solid rgba(0,0,0, 0.2);
        border-bottom: 0.5em solid transparent;
        z-index: 0;
    }

`
const StyledSpan = styled.span`
    width: fit-content;
    height: fit-content;
    font-size: ${props => props.fontSize};
    color: ${props => props.color};
    align-self: center;
    &.leftText{
        margin-right: 1em;
    }
    &.rightText{
        margin-left: 1.5em;
    }
`
const RemainingTimeBar = ({value, maxValue}) => {
    const getPercentage = (value, maxValue) => (100-(value * 100)/maxValue)
    return (
        <BarContainer>
            <ProgresBar value={getPercentage(value, maxValue)}>
                <StyledSpan 
                    className="leftText"
                    fontSize = {cssConstants.textSizes.small}
                    color = {white.hex}
                    >
                    CLOSING IN
                </StyledSpan>
            </ProgresBar>
            <StyledSpan 
                className="rightText"
                fontSize = {cssConstants.textSizes.semiMedium}
                color = {coal.hex}
                >
                {value} days
            </StyledSpan>
        </BarContainer>
    );
}

RemainingTimeBar.propTypes = {
    value: PropTypes.number,
    maxValue: PropTypes.number,
};

RemainingTimeBar.defaultProps = {
    value:20,
    maxValue: 100,
}

export default RemainingTimeBar;
