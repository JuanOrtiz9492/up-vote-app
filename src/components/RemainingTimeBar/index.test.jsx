import React from 'react';
import { render } from '@testing-library/react';
import RemainingTimeBar from '.';

describe('RemainingTimeBar Component', () => {

    it('should Match the Snapshot', () => {
        const { container } = render(<RemainingTimeBar />);
        expect(container).toMatchSnapshot();
      })
})

