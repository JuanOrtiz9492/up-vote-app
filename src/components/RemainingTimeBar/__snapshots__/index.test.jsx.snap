// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemainingTimeBar Component should Match the Snapshot 1`] = `
<div>
  <div
    class="sc-AykKC eCWvRg"
  >
    <div
      class="sc-AykKD dFcBtW"
      value="80"
    >
      <span
        class="sc-AykKE htFEyL leftText"
        color="#ffffff"
        font-size="0.8em"
      >
        CLOSING IN
      </span>
    </div>
    <span
      class="sc-AykKE kriIIj rightText"
      color="#333333"
      font-size="1.5em"
    >
      20
       days
    </span>
  </div>
</div>
`;
